Services/DataX.Flow/Flow.ManagementService/Controllers/FlowManagementController.cs

```diff
@@ -463,5 +481,42 @@ public async Task<ApiResult> SyncJobsByNames([FromBody] string[] jobNames)
             }
         }
 
+        [HttpPost]
+        [Route("job/waittilljobsrunning")]
+        public async Task<ApiResult> WaitTillJobsRunning([FromBody] WaitTillJobsRunningRequest waitRequest)
+        {
+            try
+            {
+                RolesCheck.EnsureWriter(Request, _isLocal);
+
+                var maxRetryTimes = waitRequest.MaxRetryTimes;
+                var jobNames = waitRequest.JobNames;
+                var intervalsBetweenRetriesInSeconds = waitRequest.IntervalsBetweenRetriesInSeconds;
+
+                int retryTimes = 0;
+                SparkJobFrontEnd[] jobOpResult = null;
+                while (retryTimes++ < maxRetryTimes)
+                {
+                    jobOpResult = jobNames != null && jobNames.Any() ?
+                                await _jobOperation.SyncJobStateByNames(jobNames)
+                                : await _jobOperation.SyncAllJobState();
+
+                    var startingJobs = jobOpResult.Where(r => r.JobState == JobState.Starting);
```

Comment:
In case resources donot get allocated to the spark job on HDInsight, the state changes to idle. In this case job submission needs to be retired too

Reply:
have updated the logic to restart the job state is idle, starting or failed